src/main/java/aluczak/reservationsystem/ReservationController.java

```diff
@@ -0,0 +1,178 @@
+                        HttpStatus.BAD_REQUEST);
+            } else {
+                saveInputReservation(reservationInput);
+                LocalDateTime expirationTime = getScreeningTimeFromInput(reservationInput);
+                setupSeatsId(reservationInput);
+                BigDecimal totalPrice = getTotalPrice(reservationInput);
+                return new ResponseEntity<>("Total amount to pay: " + totalPrice
+                        + " Expiration Time: " + expirationTime,
+                        HttpStatus.OK);
+            }
+        }
+    }
+
+    public void saveInputReservation(Reservation reservationInput) {
+        Optional<Movie> movieOptional = movieRestRepository.findById(reservationInput.movie.movieId);
+        movieOptional.ifPresent(movie -> reservationRepository.save(
+                new Reservation(reservationInput.firstName,
+                        reservationInput.surname,
+                        movie,
+                        reservationInput.seats)));
+    }
+
+    public LocalDateTime getScreeningTimeFromInput(Reservation reservationInput) {
+        Optional<Movie> movieOptional = movieRestRepository.findById(reservationInput.movie.movieId);
+        return movieOptional.map(value -> value.screeningDateTime).orElse(null);
+    }
+
+    public void setupSeatsId(Reservation reservationInput) {
+        long lastReservationId = reservationRepository.count();
+        Optional<Reservation> optionalReservation = reservationRepository.findById(lastReservationId);
+        Reservation reservation;
+        if (optionalReservation.isPresent()) {
+            reservation = optionalReservation.get();
+        } else {
+            throw new RuntimeException("Reservation of id " + lastReservationId + " doesn't exist.");
+        }
+        long lastSeatId = seatRepository.count();
+
+        for (int i = 0; i < reservation.seats.size(); i++) {
+
+            Optional<Seat> optionalSeat = seatRepository.findById(lastSeatId - i);
+            Seat seat;
+            if (optionalSeat.isPresent()) {
+                seat = optionalSeat.get();
+            } else {
+                throw new RuntimeException();
+            }
+            seat.setReservation(reservation);
+            seatRepository.save(seat);
+        }
+    }
+
+    public BigDecimal getTotalPrice(Reservation reservationInput) {
+        BigDecimal totalPrice = BigDecimal.valueOf(0);
+        for (Seat seat : reservationInput.seats) {
+            BigDecimal price = ticketRepository.findByType(seat.ticketType).price;
+            totalPrice = totalPrice.add(price);
+        }
+        return totalPrice;
+    }
+
+    private String validateSeats(Reservation reservationInput) {
+
+        Optional<Movie> optionalMovie = movieRestRepository.findById(reservationInput.movie.movieId);
+        Movie movie;
+        if (optionalMovie.isPresent()) {
+            movie = optionalMovie.get();
+        } else {
+            throw new RuntimeException();
```

Comment:
Czy to dobry wyjątek? I czy nie brakuje tu opisu?

Reply:
Faktycznie, może lepiej byłoby użyć bardziej opisowego wyjątku np. NoSuchElementException z message: No such movie `[movieId]`.